test/unit/generator/create-filename.test.js

```diff
@@ -124,14 +127,12 @@ describe('create filename', () => {
   })
 
   test('removes spaces if statement', () => {
-    mockStatement.scheme.shortName = 'My Scheme'
-    const result = getFilename(mockStatement, timestamp, STATEMENT)
-    expect(result).toContain('MyScheme')
+    const result = getFilename(mockSchedule, timestamp, STATEMENT)
+    expect(result).toBe('FFC_PaymentStatement_SFI_2022_1234567890_2022080515301012.pdf')
   })
 
   test('removes spaces if schedule', () => {
-    mockSchedule.scheme.shortName = 'My Scheme'
     const result = getFilename(mockSchedule, timestamp, SCHEDULE)
-    expect(result).toContain('MyScheme')
+    expect(result).toBe('FFC_PaymentSchedule_SFI_2022_1234567890_2022080515301012.pdf')
```

Comment:
There should be  tests to prove the following:

1. No error message when filename composition is valid.
2. Expected filename is created when filename composition is valid
3. There is error message when filename composition is not valid.
4. The expected filename is not created when filename is not valid.

Reply:
I have added a test that checks that an incorrect filename creation results in the throw Error as well as the examples above that are checking that the final filename creation without spaces is in the correct format.